File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.py
def solution(s):
    idx = 0
    stack = []
    while(idx < len(s)):
        if stack and stack[-1] == s[idx]:
            stack.pop()
        else:
            stack.append(s[idx])
        idx+=1
    
    return 0 if len(stack)>0 else 1 

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.js
function solution(s){
    const stack = []
    let idx = 0
    for(let ch of s){
        if(stack.length && stack[stack.length-1] == ch){
            stack.pop();
        }else{
            stack.push(ch);
        }
    }
    return stack.length == 0 ? 1 : 0
}